hoomd/hpmc/update.py

```diff
@@ -300,160 +301,49 @@ class muvt(_updater):
-            raise RuntimeError("Error initializing update.muvt");
-
-        cpp_transfer_types = _hoomd.std_vector_uint();
-        for t in transfer_types:
-            if t not in type_list:
-                hoomd.context.current.device.cpp_msg.error("Trying to transfer unknown type " + str(t) + "\n");
-                raise RuntimeError("Error setting muVT parameters");
-            else:
-                type_id = hoomd.context.current.system_definition.getParticleData().getTypeByName(t);
-
-            cpp_transfer_types.append(type_id)
-
-        self.cpp_updater.setTransferTypes(cpp_transfer_types)
-
-    def set_fugacity(self, type, fugacity):
-        R""" Change muVT fugacities.
-
-        Args:
-            type (str): Particle type to set parameters for
-            fugacity (float): Fugacity of this particle type (dimension of volume^-1)
-
-        Example::
-
-            muvt = hpmc.update.muvt(mc, period=10)
-            muvt.set_fugacity(type='A', fugacity=1.23)
-            variant = hoomd.variant.linear_interp(points=[(0,1e1), (1e5, 4.56)])
-            muvt.set_fugacity(type='A', fugacity=variant)
-
-        """
-        self.check_initialization();
-
-        if self.gibbs:
-            raise RuntimeError("Gibbs ensemble does not support setting the fugacity.\n");
-
-        # get a list of types from the particle data
-        ntypes = hoomd.context.current.system_definition.getParticleData().getNTypes();
-        type_list = [];
-        for i in range(0,ntypes):
-            type_list.append(hoomd.context.current.system_definition.getParticleData().getNameByType(i));
-
-        if type not in type_list:
-            hoomd.context.current.device.cpp_msg.error("Trying to set fugacity for unknown type " + str(type) + "\n");
-            raise RuntimeError("Error setting muVT parameters");
-        else:
-            type_id = hoomd.context.current.system_definition.getParticleData().getTypeByName(type);
-
-        fugacity_variant = hoomd.variant._setup_variant_input(fugacity);
-        self.cpp_updater.setFugacity(type_id, fugacity_variant.cpp_variant);
-
-    def set_params(self, dV=None, volume_move_probability=None, n_trial=None):
-        R""" Set muVT parameters.
-
-        Args:
-            dV (float): (if set) Set volume rescaling factor (dimensionless)
-            volume_move_probability (float): (if set) In the Gibbs ensemble, set the
-                probability of volume moves (other moves are exchange/transfer moves).
-            n_trial (int): (if set) Number of re-insertion attempts per depletant
-
-        Example::
+    def __init__(self, seed, transfer_types=None, ngibbs=1, max_volume_rescale=0.1,
+        move_ratio=0.5, trigger=1):
+        super().__init__(trigger)
 
-            muvt = hpmc.update.muvt(mc, period = 10)
-            muvt.set_params(dV=0.1)
-            muvt.set_params(n_trial=2)
-            muvt.set_params(volume_move_probability=0.05)
+        self.ngibbs = int(ngibbs)
```

Comment:
Is there a reason that `ngibbs` isn't in the `param_dict`?

Reply:
yes, it doesn't really make sense to dynamically reconfigure the MPI partition layout during execution..